shared/KOSocket.cpp

```diff
@@ -139,23 +141,27 @@ bool KOSocket::DecryptPacket(uint8 *in_stream, Packet & pkt)
 
 bool KOSocket::Send(Packet * pkt) 
 {
-	if (!IsConnected() || pkt->size() + 1 > GetWriteBuffer().GetAllocatedSize())
+	if (!IsConnected()
+		|| pkt->size() + 7 > MAX_PACKET_SEND_SIZE)
```

Comment:
`GetWriteBuffer().GetAllocatedSize()` feels more preferable rather than using a constant. Why comparing it to `MAX_PACKET_SEND_SIZE` is necessary?
Also why `pkt->size() + 7`? `+1` somewhat makes more sense due to a null-terminated buffer, but `+7` raises questions. Can you explain?